### PR TITLE
SSR to webgpu

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15420/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15420/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15420/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15420/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15420/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15420/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15420/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15420/merge#BCU1XR#0

File: packages/dev/core/src/ShadersWGSL/ShadersInclude/screenSpaceRayTrace.fx [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+
+    // Screen-space endpoints
+    var P0: vec2f = H0.xy * k0;
+    var P1: vec2f = H1.xy * k1;
+
+#ifdef SSRAYTRACE_CLIP_TO_FRUSTUM
+    var xMax: f32 = csZBufferSize.x - 0.5;
+    var xMin = 0.5;
+    var yMax = csZBufferSize.y - 0.5;
+    var yMin = 0.5;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/ShadersInclude/screenSpaceRayTrace.fx:112]: type is missing. is it intended?

File: packages/dev/core/src/ShadersWGSL/ShadersInclude/screenSpaceRayTrace.fx [hunk 2]
```diff
@@ -0,0 +1,352 @@
+// Screen Space Ray Tracing based on:
+// * http://casual-effects.blogspot.com/2014/08/screen-space-ray-tracing.html
+// * https://sourceforge.net/p/g3d/code/HEAD/tree/G3D10/data-files/shader/screenSpaceRayTrace.glsl
+// * https://github.com/kode80/kode80SSR
+// Adapted to Babylon.js case where the camera space coordinate system is left-handed (visible geometries in this space have positive Z values)
+
+fn distanceSquared(a: vec2f, b: vec2f) -> f32 { 
+    var temp = a - b;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/ShadersInclude/screenSpaceRayTrace.fx:8]: same here

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15420/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/ShadersWGSL/screenSpaceReflection2BlurCombiner.fragment.fx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+#ifdef SSR_INPUT_IS_GAMMA_SPACE
+    color = toLinearSpaceVec4(color);
+#endif
+
+#ifdef SSR_BLEND_WITH_FRESNEL
+    var texSize: vec2f =  vec2f(textureDimensions(depthSampler, 0));
+
+    var csNormal: vec3f = textureLoad(normalSampler, vec2<i32>(input.vUV * texSize), 0).xyz;
+    var depth: f32 = textureLoad(depthSampler, vec2<i32>(input.vUV * texSize), 0).r;
+    var csPosition: vec3f = computeViewPosFromUVDepth(input.vUV, depth, projection, invProjectionMatrix);
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/screenSpaceReflection2BlurCombiner.fragment.fx:54]:
```suggestion
    var csPosition: vec3f = computeViewPosFromUVDepth(input.vUV, depth, uniforms.projection, uniforms.invProjectionMatrix);
```

File: packages/dev/core/src/ShadersWGSL/ShadersInclude/screenSpaceRayTrace.fx (hunk 1 above)
Comment (deltakosh) [reply packages/dev/core/src/ShadersWGSL/ShadersInclude/screenSpaceRayTrace.fx:112]: Yes it is inferred

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15420/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15420/merge/testResults/webgpuplaywright/index.html

**Pull request merged**